provider/openstack/networking.go

```diff
@@ -259,19 +270,30 @@ func (n *NeutronNetworking) ResolveNetwork(name string) (string, error) {
 	return resolveNeutronNetwork(n.env.neutron(), name)
 }
 
+// networkNameFilter returns a neutron.Filter to match Neutron Networks with
+// the exact given name.
+func networkNameFilter(name string) *neutron.Filter {
+	filter := neutron.NewFilter()
+	filter.Set(neutron.FilterNetwork, fmt.Sprintf("^%s$", name))
+	return filter
+}
+
 func resolveNeutronNetwork(neutron *neutron.Client, name string) (string, error) {
 	if utils.IsValidUUIDString(name) {
 		return name, nil
 	}
-	var networkIds []string
-	networks, err := neutron.ListNetworksV2()
+	// TODO (hml) 2017-03-17 related to #1654144
```

Comment:
can't we just pass in a bool to resolveNeutronNetwork saying whether or not it's external? the call in getExternalNeutronNetworksByAZ can pass in true, and the call in ResolveNetwork can pass in false?